components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ThrottlingApiServiceImpl.java

```diff
@@ -907,6 +997,323 @@ public Response throttlingPoliciesCustomRuleIdDelete(String ruleId,  MessageCont
+                try {
+                    //only super tenant is allowed to access global policies/custom rules
+                    checkTenantDomainForCustomRules();
+                    //This will give PolicyNotFoundException if there's no policy exists with UUID
+                    GlobalPolicy globalPolicy = apiProvider.getGlobalPolicy(policyName);
+                    if (globalPolicy != null) {
+                        CustomRuleDTO policyDTO = GlobalThrottlePolicyMappingUtil.fromGlobalThrottlePolicyToDTO(
+                                globalPolicy);
+                        exportPolicy.data(policyDTO);
+                        exportPolicy.subtype(RestApiConstants.RESOURCE_CUSTOM_RULE);
+                        return Response.ok().entity(exportPolicy).build();
+                    } else if (!type.equals(StringUtils.EMPTY)) {
+                        RestApiUtil.handleResourceNotFoundError(
+                                RestApiConstants.RESOURCE_CUSTOM_RULE + " not found by the name " + policyName, log);
+                    }
+                } catch (APIManagementException e) {
+                    String errorMessage = "Error while retrieving Custom policy: " + policyName;
+                    RestApiUtil.handleInternalServerError(errorMessage, e, log);
+                }
+            }
+            RestApiUtil.handleResourceNotFoundError("No throttle policy found by the name " + policyName, log);
+        } catch (APIManagementException | ParseException e) {
+            String errorMessage = "Error while retrieving throttling policy. Name : " + policyName;
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        }
+        return null;
+    }
+
+    /**
+     * Returns the ExportThrottlePolicyDTO by reading the file from input stream
+     *
+     * @param uploadedInputStream Input stream from the REST request
+     * @param fileDetail          Details of the file received via InputStream
+     * @return ExportThrottlePolicyDTO of the file to be imported
+     */
+    public static ExportThrottlePolicyDTO getImportedPolicy(InputStream uploadedInputStream, Attachment fileDetail)
+            throws ParseException, APIImportExportException, IOException {
+        File importFolder = CommonUtil.createTempDirectory(null);
+        String uploadFileName = fileDetail.getContentDisposition().getFilename();
+        String fileType = (uploadFileName.contains(ImportExportConstants.YAML_EXTENSION)) ?
+                PolicyConstants.EXPORT_POLICY_TYPE_YAML :
+                PolicyConstants.EXPORT_POLICY_TYPE_JSON;
+        String absolutePath = importFolder.getAbsolutePath() + File.separator + uploadFileName;
+        File targetFile = new File(absolutePath);
+        FileUtils.copyInputStreamToFile(uploadedInputStream, targetFile);
+        return preprocessImportedArtifact(absolutePath, fileType);
+    }
+
+    /**
+     * Preprocesses the either yaml or json file into the ExportThrottlePolicyDTO
```

Comment:
remove 'the'

Reply:
Fixed with 30d0ecb9c29c0024fa182ba597c8eda4c128c451